src/Sulu/Component/Content/Compat/Serializer/PageBridgeHandler.php

```diff
@@ -60,54 +58,61 @@ public static function getSubscribingMethods()
     }
 
     /**
-     * @param JsonSerializationVisitor $visitor
-     * @param PageBridge               $bridge
-     * @param array                    $type
-     * @param Context                  $context
+     * @param VisitorInterface $visitor
+     * @param PageBridge $bridge
+     * @param array $type
+     * @param Context $context
      */
     public function doSerialize(
-        JsonSerializationVisitor $visitor,
+        VisitorInterface $visitor,
         PageBridge $bridge,
         array $type,
         Context $context
     ) {
-        $refl = new \ReflectionClass(PageBridge::class);
-        $documentProperty = $refl->getProperty('document');
-        $structureProperty = $refl->getProperty('structure');
-        $documentProperty->setAccessible(true);
-        $structureProperty->setAccessible(true);
+        $document = $bridge->getDocument();
+        $structure = $bridge->getStructure();
 
-        $document = $documentProperty->getValue($bridge);
-        $structure = $structureProperty->getValue($bridge);
+        $data = [];
+        foreach ($bridge->getProperties(true) as $property) {
+            $data[$property->getName()] = $property->getValue();
+        }
```

Comment:
Why is this necessary? Shouldn't the document also have this information? Then I would try to implement that correctly in the document serialization, since the PageBridge will be gone in the long term.